apps/site/src/_pages/docs/3_spec/03_hook-service.mdx

```diff
@@ -0,0 +1,120 @@
+
+This document (version 0.1) defines a **hook service** according to the terms set out in the core specification.
+
+Under the hook service, a block can invite embedding application to take over the rendering of an element within the block (the **hook**), allowing embedding applications to provide users with application-specific **views** of specific data types without the block needing to know about them.
+
+The service is designed to meet the following aims:
+
+- embedding applications providing a consistent 'look and feel' across blocks for displaying or editing values of a particular type (e.g. rich text or image editing)
+- allowing blocks to be portable across applications which use bespoke systems or data structures for displaying or editing values of a particular type (e.g. there is no universal “rich text” representation for blocks to parse and produce)
+
+The service meets these aims by allowing blocks to delegate visualizing and editing functionality upwards to the embedding application.
+
+The hook service is dependent on [the graph service](https://blockprotocol.org/docs/spec/graph-service), as it relies on the existence of **entities** – use of the graph service is a prerequisite for use of the hook service.
+
+## Glossary
+
+**Hook:** an injection point blocks provide to embedding applications to allow them to optionally render or modify already rendered views associated with a property for an entity.
+
+**View:** the elements the embedding application renders inside the hook (and associated functionality). A view may display read-only data or present editable data at the application’s discretion.
+
+## Establishing hooks
+
+### <Hidden>Establishing hooks </Hidden>Summary
+
+Blocks initiate hooks by asking embedding applications to take over a DOM node (the hook) and render the application’s preferred view of data of a specified type.
+
+### <Hidden>Establishing hooks </Hidden>Specification
+
+To request a view is rendered into a hook, a block MUST send a `hook` [message](https://www.notion.so/Hook-service-spec-1fcc185dc67c4ce7ace06878a144977d). The value for `hookId` MUST be `null` when a block first establishes a hook.
+
+If the embedding application **supports** the requested hook type:
+
+- it MUST:
+  - render a view informed by the contents of the `hook` message:
+    - of the appropriate `type`
+    - by manipulating the specified `node` or by attaching elements within it
+    - using data stored at the specified `path` on the entity associated with the specified `entityId`
+  - send a `hookResponse` message specifying a `hookId` to identify the hook with in future
+- It MUST NOT destroy the node or move it within the DOM.
+
+If the embedding application **does not** **support** the requested hook type, it MUST respond with an error with a code of `NOT_IMPLEMENTED`.
+
+If any of the required data is missing from the `hook` message, the emebdding application MUST respond with an error with a code of `INVALID_INPUT`.
```

Comment:
typo

Reply:
Thank you – https://github.com/blockprotocol/blockprotocol/pull/676/commits/6a2b9e65ad1deb6d0251843f09f205debb35d4b9